### PR TITLE
Fix some new_filled_column kwargs

### PR DESCRIPTION
**Pull request by benjello**
The `new_filled_column` function takes several kwargs.

Some of them – `set_input`, `calculate_output` and `default` are not handled correctly.

This is a fix for this problem.

### CODE REVIEW

File: openfisca_core/formulas.py [hunk 1]
```diff
@@ -676,9 +676,10 @@ def neutralize_column(column):
         )
 
 
-def new_filled_column(base_function = UnboundLocalError, calculate_output = UnboundLocalError,
+def new_filled_column(__doc__ = None, __module__ = None,
+        base_function = UnboundLocalError, calculate_output = UnboundLocalError,
         cerfa_field = UnboundLocalError, column = UnboundLocalError, comments = UnboundLocalError,
-        __doc__ = None, __module__ = None,
+        default = UnboundLocalError,
```
Comment (cbenz) [inline openfisca_core/formulas.py]: Did you remove `__doc__` and `__module__` kwargs default values intentionally?

File: openfisca_core/formulas.py [hunk 2]
```diff
@@ -694,7 +695,7 @@ def new_filled_column(base_function = UnboundLocalError, calculate_output = Unbo
     assert isinstance(name, unicode)
 
     if calculate_output is UnboundLocalError:
-        calculate_output = None if reference_column is None else reference_column.formula_class.calculate_output
+        calculate_output = None if reference_column is None else reference_column.formula_class.calculate_output.im_func
```
Comment (cbenz) [inline openfisca_core/formulas.py]: What is `.im_func`? Is this change related to the new `default` kwarg?

File: setup.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 setup(
     name = 'OpenFisca-Core',
-    version = '4.2.1',
+    version = '4.2.2',
```
Comment (cbenz) [inline setup.py]: Perhaps a minor bump?

File: openfisca_core/formulas.py (hunk 2 above)
Comment (benjello) [inline openfisca_core/formulas.py]: It is the implementation function and not the method related to the reference_column_formula_calss. Ask @eraviart for more technical details ! ;-)

File: openfisca_core/formulas.py (hunk 1 above)
Comment (benjello) [inline openfisca_core/formulas.py]: Nope just alphabetic sorting

File: setup.py (hunk 3 above)
Comment (benjello) [inline setup.py]: The goal of this PR was to fix a bug when neutralizing a updating a column to make sure the set_input and default value are correctly transmitted to the update variable.

**Comment by benjello** [comment]
@cbenz : is it clearer ?

File: openfisca_core/formulas.py (hunk 2 above)
Comment (fpagnoux) [inline openfisca_core/formulas.py]: I confirm, this is the way to get the pure function out of a method.

File: openfisca_core/formulas.py (hunk 1 above)
Comment (fpagnoux) [inline openfisca_core/formulas.py]: Are we going to pass a `default` argument to the `new_filled_column` method ?

Comment (benjello) [inline openfisca_core/formulas.py]: If I recall well: we had a problem when neutralizing column that had a default value which was not the parent Column default value. @eraviart can provide you more details ;-)

**Comment by cbenz** [comment]
@benjello I changed the description of the PR for you ;-)

**Comment by fpagnoux** [comment]
@benjello : 
Do you know about `git rebase` ? Upstream merges make the git topology hard to read. For changes such as this PR, I think a rebase would be easy.

**Comment by fpagnoux** [comment]
There is an ongoing discussion about in in `#tech` ;)

**Comment by benjello** [comment]
I let you merge it then !

**Comment by benjello** [comment]
I will merge it.
